core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -56,7 +56,7 @@
 public class BaseTransaction implements Transaction {
   private static final Logger LOG = LoggerFactory.getLogger(BaseTransaction.class);
 
-  enum TransactionType {
+  protected enum TransactionType {
```

Comment:
Rather than exposing this, could we add public factory methods? `newReplaceTransaction(...)` for example?

Reply:
I also moved the constructor method back to package private, as TransactionType is part of its constructor parameter to keep consistent visibility. I think it's a good compromise to only vend out new transactions in factory class located in `org.apache.iceberg` package